### ISSUE TITLE
SaveButton disabled behavior inconsistent with custom toolbar documentation

### ISSUE THREAD

**Issue by sebitoelcheater** [marmelab/react-admin#5220]
**What you were expecting:**
After I use a custom toolbar in an Edit component , the save button should keep disabled until the form has any modifications or it's dirty.

**What happened instead:**
The save button keeps enabled.

**Steps to reproduce:**
```
const CustomToolbar = props => (
  <Toolbar {...props} >
    <SaveButton/>
  </Toolbar>
);

export default props => (
  <Edit undoable={false} {...props} initialValues={{"company.avatar.original": ''}}>
    <TabbedForm redirect={"edit"} toolbar={<CustomToolbar/>}>
      <FormTab redirect={"edit"} label="Personal data">
        <TextInput source="first_name"/>
      </FormTab>
    </TabbedForm>
  </Edit>
);
```

**Environment**

* React-admin version: ^3.7.1
* Last version that did not exhibit the issue (if applicable): 
* React version: ^16.13.1
* Browser: chrome
* Stack trace (in case of a JS error):

**Comment by djhi** [comment]
Thanks for reporting this. If you are able to illustrate the bug or feature request with an example, please provide a sample application by forking the following CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple).

**Comment by wattry** [comment]
We have noticed this too. I'd rephrase the issue as such: "SaveButton disabled behavior inconsistent with custom toolbar documentation". In the documentation it is states 
> [By default the \<SaveButton\> is disabled when the form is pristine.](https://marmelab.com/react-admin/CreateEdit.html#toolbar).

When using a custom Toolbar and react-admin out of the box, the save button does not seem to follow the example provided in the codesandbox.

Here is an example: https://codesandbox.io/s/polished-sun-xqkcm?file=/src/Posts.js

**Comment by FernandoVelcic** [comment]
Hello I want to reopen this ticket because the problem is not the documentation only, it's impossible to disable the save button (when the form is invalid) with a custom toolbar

`
const customToolbar = toolbarProps => {
    return (
        <Toolbar {...toolbarProps} >
            <SaveButton disabled={!toolbarProps.pristine && !toolbarProps.validating} invalid={toolbarProps.invalid} />
        </Toolbar>
    )
};
`
pristine is updating correctly but validating prop keep the same value even if the form is valid or not.

Thanks.
